R/code_dependencies.R

```diff
@@ -1,65 +1,56 @@
-## TODO: According to JJ, there is support for doing this sort of
-## thing in a DTL package - possibly called 'codedeps'.  Probably
-## worth finding out how that works and seeing if I can make this
-## faster and more reliable.
-code_dependencies <- function(f, hide_errors=TRUE) {
+code_dependencies <- function(f, hide_error=TRUE) {
   env <- environment(f)
-  ## Exclude these:
+  ## Exclude these as they shadow variables:
   args <- names(formals(f))
 
-  leaf <- function(e, w) {
-    if (!is.symbol(e)) { # A literal of some type
-      return()
-    }
-    e_name <- deparse(e)
-
-    ## Shadowed by argument:
-    ## TODO: Nested function definitions will not work correctly
-    ## here; we'd need to switch the "active" function.  Getting this
-    ## wrong is most likely to cause false positives.
-    if (e_name %in% args) {
-      return()
-    }
+  ## This is what we build up:
+  ret <- list(functions=character(0),
+              packages=character(0))
 
-    ## TODO: Things like get() will totally confuse this.
-    if (!exists(e_name, env, inherits=FALSE) &&
-        ((!exists(e_name, env)       || # local variable, probably
-          is_active_binding(e_name)))) {  # using remake active bindings
```

Comment:
Oh my. Get well soon.

Tests work, but I "had" to use a function inside a function in my data analysis project, which failed with the new code. This PR fixes that. I'm opening a new issue pointing to my review, so that we can merge as is